app/models/user.rb

```diff
@@ -57,6 +65,25 @@ def send_password_reset_email
     UserMailer.password_reset(self).deliver_now
   end
 
+  def follow other_user
+    following << other_user
+  end
+
+  def unfollow other_user
+    following.delete other_user
+  end
+
+  def following? other_user
+    following.include? other_user
+  end
+
+  def feed
+    following_ids = "SELECT followed_id FROM relationships
+      WHERE  follower_id = :user_id"
+    Micropost.where("user_id IN (#{following_ids})
```

Comment:
where("user_id IN (#{following_ids}) OR user_id = :user_id", user_id: id)
cái này viết vào scope nhé